netket/utils/numbers.py

```diff
@@ -16,6 +16,13 @@ def dtype(x: Array):  # noqa: F811, E0102
     return x.dtype
 
 
+@dispatch
```

Comment:
much happier now, so happy. but why there is still this dispatch over here?

Reply:
Because you shan't be too happy. Just happier.

Reply:
❤️❤️❤️

Reply:
because we are supporting python numbers and everything else.